### PR TITLE
Manage lifetime of m_pStyle using a std::unique_ptr

### PR DESCRIPTION
**Pull request by daschuer**
This fixes a memory leak

### CODE REVIEW

**Comment by ronso0** [comment]
Why does this QStyle  need to be a member not a local variable in the first place? It's created and used only in the constructor & init.

**Comment by uklotzde** [comment]
I agree with @ronso0 Using a unique_ptr is pointless here. It has the same lifetime as a value member.

**Comment by uklotzde** [comment]
The class must be non-copyable in both cases.

**Comment by daschuer** [comment]
> Why does this QStyle need to be a member not a local variable in the first place? It's created and used only in the constructor & init.

It is borrowed by the widgets which hold a copy of the pointer. If we would delete it, the pointer returned by QWidget::style() becomes dangling. 

https://doc.qt.io/qt-5/qwidget.html#setStyle

> I agree with @ronso0 Using a unique_ptr is pointless here. It has the same lifetime as a value member.
QStyle is not copy-able. We simply have only the pointer created by QStyleFactory::create(). 


> The class must be non-copyable in both cases. With a value member it must also be non-movable.

This is already the case via the base class: 
 
https://github.com/qt/qtbase/blob/4ee4fc18b4067b90efa46ca9baba74f53b54d9ec/src/widgets/dialogs/qdialog.h#L118

**Comment by uklotzde** [comment]
> > Why does this QStyle need to be a member not a local variable in the first place? It's created and used only in the constructor & init.

Ok. I didn't check that the instance is created by a factory that returns a pointer. Then using a unique_ptr is the way to go here.

File: src/dialog/dlgreplacecuecolor.h [hunk 1]
```diff
@@ -66,6 +67,6 @@ class DlgReplaceCueColor : public QDialog, public Ui::DlgReplaceCueColor {
     parented_ptr<WColorPickerAction> m_pCurrentColorPickerAction;
     mixxx::RgbColor::optional_t m_lastAutoSetNewColor;
     mixxx::RgbColor::optional_t m_lastAutoSetCurrentColor;
-    QStyle* m_pStyle;
+    std::unique_ptr<QStyle> m_pStyle;
```
Comment (uklotzde) [inline src/dialog/dlgreplacecuecolor.h:70]: This member should be `const` to prevent that the memory is freed after handing it to the widgets.

**Review by uklotzde — APPROVED** [review]

File: src/dialog/dlgreplacecuecolor.h (hunk 1 above)
Comment (Swiftb0y) [reply src/dialog/dlgreplacecuecolor.h:70]: How does that prevent freeing the memory? IMO instead the pointer should be explicitly `.release()`d at definite last use.

**Review by Swiftb0y — APPROVED** [review]

**Pull request merged**